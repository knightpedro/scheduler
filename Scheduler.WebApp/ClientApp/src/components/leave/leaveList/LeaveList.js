import React, { useMemo } from "react";
import { Link, generatePath } from "react-router-dom";
import Routes from "../../../routes";
import moment from "moment";
import { BasicTable, SelectColumnFilter } from "../../common/tables";

const TIME_FORMAT = "HH:mm DD/MM/YYYY";

const LeaveList = ({ leave }) => {
  const createPath = id => generatePath(Routes.leave.EDIT, { id });

  const columns = useMemo(
    () => [
      {
        Header: "Type",
        accessor: "type",
        Cell: ({ row }) => (
          <Link to={createPath(row.original.id)}>{row.values.type}</Link>
        ),
        Filter: SelectColumnFilter
      },
      {
        Header: "Start",
        accessor: "start",
        className: "text-center",
        sortType: "momentSort",
        disableFilters: true,
        Cell: ({ cell: { value } }) => value.format(TIME_FORMAT)
      },
      {
        Header: "End",
        accessor: "end",
        className: "text-center",
        sortType: "momentSort",
        disableFilters: true,
        Cell: ({ cell: { value } }) => value.format(TIME_FORMAT)
      }
    ],
    []
  );

  const data = useMemo(
    () =>
      leave.map(l => {
        return {
          id: l.id,
          type: l.description,
          start: moment(l.start),
          end: moment(l.end)
        };
      }),
    [leave]
  );

  const initialTableState = useMemo(
    () => ({
      sortBy: [
        {
          id: "start",
          desc: true
        }
      ]
    }),
    []
  );

  return (
    <BasicTable
      columns={columns}
      data={data}
      initialState={initialTableState}
    />
  );
};

export default LeaveList;
