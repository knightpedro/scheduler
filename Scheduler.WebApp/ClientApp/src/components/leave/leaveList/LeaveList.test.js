import React from "react";
import { shallow } from "enzyme";
import LeaveList from "./LeaveList";

const leaveStub = [
    {
        id: 1,
        description: "Annual Leave",
        start: "2020-02-29T19:00:00.0000000Z",
        end: "2020-02-29T23:00:00.0000000Z",
    },
];

describe("LeaveList", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<LeaveList leave={leaveStub} />);
        expect(wrapper).toMatchSnapshot();
    });
});
