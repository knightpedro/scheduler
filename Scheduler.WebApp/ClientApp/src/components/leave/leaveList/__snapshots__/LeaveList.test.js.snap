// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveList renders correctly 1`] = `
<BasicTable
  columns={
    Array [
      Object {
        "Cell": [Function],
        "Filter": [Function],
        "Header": "Type",
        "accessor": "type",
      },
      Object {
        "Cell": [Function],
        "Header": "Start",
        "accessor": "start",
        "className": "text-center",
        "disableFilters": true,
        "sortType": "momentSort",
      },
      Object {
        "Cell": [Function],
        "Header": "End",
        "accessor": "end",
        "className": "text-center",
        "disableFilters": true,
        "sortType": "momentSort",
      },
    ]
  }
  data={
    Array [
      Object {
        "end": "2020-02-29T23:00:00.000Z",
        "id": 1,
        "start": "2020-02-29T19:00:00.000Z",
        "type": "Annual Leave",
      },
    ]
  }
  initialState={
    Object {
      "sortBy": Array [
        Object {
          "desc": true,
          "id": "start",
        },
      ],
    }
  }
/>
`;
